luxtronik/__init__.py

```diff
@@ -24,28 +54,59 @@ def __init__(self, host, port, safe=True):
         self.visibilities = Visibilities()
         self.read()
 
-    def _connect(self):
-        self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        self._socket.connect((self._host, self._port))
-        LOGGER.info("Connected to Luxtronik heatpump %s:%s", self._host, self._port)
-
-    def _disconnect(self):
-        self._socket.close()
-        LOGGER.info(
-            "Disconnected from Luxtronik heatpump  %s:%s", self._host, self._port
-        )
+    def __del__(self):
+        if self._socket is not None:
+            if not is_socket_closed(self._socket):
+                self._socket.close()
+            self._socket = None
+            LOGGER.info(
+                "Disconnected from Luxtronik heatpump %s:%s", self._host, self._port
+            )
 
     def read(self):
         """Read data from heatpump."""
-        self._connect()
+        self._read_write(write=False)
+
+    def write(self):
+        """Write parameter to heatpump."""
+        self._read_write(write=True)
+
+    def _read_write(self, write=False):
```

Comment:
I'm not particularly good with names and the change itself looks good. Just wondering whether this function could/should be renamed to something like `_read_after_write`.

To me `_read_write()` doesn't express what is actually happening, also it should be more like `_write_read()`, since values are first written, then read back :-).